cmd/kubeadm/app/constants/constants.go

```diff
@@ -644,31 +644,23 @@ func GetDNSIP(svcSubnetList string, isDualStack bool) (net.IP, error) {
 }
 
 // GetKubernetesServiceCIDR returns the default Service CIDR for the Kubernetes internal service
-func GetKubernetesServiceCIDR(svcSubnetList string, isDualStack bool) (*net.IPNet, error) {
-	if isDualStack {
-		// The default service address family for the cluster is the address family of the first
-		// service cluster IP range configured via the `--service-cluster-ip-range` flag
-		// of the kube-controller-manager and kube-apiserver.
-		svcSubnets, err := netutils.ParseCIDRs(strings.Split(svcSubnetList, ","))
-		if err != nil {
-			return nil, errors.Wrapf(err, "unable to parse ServiceSubnet %v", svcSubnetList)
-		}
-		if len(svcSubnets) == 0 {
-			return nil, errors.New("received empty ServiceSubnet for dual-stack")
-		}
-		return svcSubnets[0], nil
-	}
-	// internal IP address for the API server
-	_, svcSubnet, err := netutils.ParseCIDRSloppy(svcSubnetList)
+func GetKubernetesServiceCIDR(svcSubnetList string) (*net.IPNet, error) {
+	// The default service address family for the cluster is the address family of the first
+	// service cluster IP range configured via the `--service-cluster-ip-range` flag
+	// of the kube-controller-manager and kube-apiserver.
+	svcSubnets, err := netutils.ParseCIDRs(strings.Split(svcSubnetList, ","))
 	if err != nil {
 		return nil, errors.Wrapf(err, "unable to parse ServiceSubnet %v", svcSubnetList)
 	}
-	return svcSubnet, nil
+	if len(svcSubnets) == 0 {
+		return nil, errors.New("received empty ServiceSubnet for dual-stack")
```

Comment:
now that dual stack is implicit, should we just say?

```suggestion
		return nil, errors.New("received empty ServiceSubnet")
```

Reply:
Updated